scripts/ci_harbor_artifact_count.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+# This script is created to determine if a harbor image exists in the repository
+# It will print the number of artifacts that exist for a particualar image.
+LOG_FILE="harbor_check.log"
+echo "CI Harbor Check Log File" > "$LOG_FILE"
+
+if [ -z "${DATAFED_HARBOR_REGISTRY}" ]
+then
+  local_DATAFED_HARBOR_REGISTRY=""
+else
+  local_DATAFED_HARBOR_REGISTRY=$(printenv DATAFED_HARBOR_REGISTRY)
+fi
+
+local_DATAFED_HARBOR_URL="https://$local_DATAFED_HARBOR_REGISTRY"
+
+if [ -z "${DATAFED_HARBOR_PROJECT}" ]
+then
+  local_DATAFED_HARBOR_PROJECT="datafed"
+else
+  local_DATAFED_HARBOR_PROJECT=$(printenv DATAFED_HARBOR_PROJECT)
+fi
+
+# Should be the name of the image
+if [ -z "${DATAFED_HARBOR_REPOSITORY}" ]
+then
+  local_DATAFED_HARBOR_REPOSITORY=""
+else
+  local_DATAFED_HARBOR_REPOSITORY=$(printenv DATAFED_HARBOR_REPOSITORY)
+fi
+
+if [ -z "${DATAFED_HARBOR_IMAGE_TAG}" ]
+then
+  local_DATAFED_HARBOR_IMAGE_TAG="latest"
+else
+  local_DATAFED_HARBOR_IMAGE_TAG=$(printenv DATAFED_HARBOR_IMAGE_TAG)
+fi
+
+if [ -z "${DATAFED_HARBOR_USERNAME}" ]
+then
+  local_DATAFED_HARBOR_USERNAME=""
+else
+  local_DATAFED_HARBOR_USERNAME=$(printenv DATAFED_HARBOR_USERNAME)
+fi
+
+if [ -z "${DATAFED_HARBOR_PASSWORD}" ]
+then
+  local_DATAFED_HARBOR_PASSWORD=""
+else
+  local_DATAFED_HARBOR_PASSWORD=$(printenv DATAFED_HARBOR_PASSWORD)
+fi
+
+if [ -z "$local_DATAFED_HARBOR_REGISTRY" ]
+then
+  echo "The DATAFED Harbor Registry has not been defined." >> "$LOG_FILE"
+  exit 1
+fi
+
+if [ -z "$local_DATAFED_HARBOR_REPOSITORY" ]
+then
+  echo "The DATAFED Harbor Repository has not been defined." >> "$LOG_FILE"
+  exit 1
+fi
+
+if [ -z "$local_DATAFED_HARBOR_USERNAME" ]
+then
+  echo "The DATAFED Harbor Username has not been defined." >> "$LOG_FILE"
+  exit 1
+fi
+
+if [ -z "$local_DATAFED_HARBOR_PASSWORD" ]
+then
+  echo "The DATAFED Harbor Password has not been defined." >> "$LOG_FILE"
+  exit 1
+fi
```

Comment:
Would it be possible to make this a for loop?

```sh
local_DATAFED_HARBOR_REGISTRY=${DATAFED_HARBOR_REGISTRY:-""}
local_DATAFED_HARBOR_REPOSITORY=${DATAFED_HARBOR_REPOSITORY:-""}
local_DATAFED_HARBOR_USERNAME=${DATAFED_HARBOR_USERNAME:-""}
local_DATAFED_HARBOR_PASSWORD=${DATAFED_HARBOR_PASSWORD:-""}

# Check for required variables
for var in local_DATAFED_HARBOR_REGISTRY local_DATAFED_HARBOR_REPOSITORY local_DATAFED_HARBOR_USERNAME local_DATAFED_HARBOR_PASSWORD; do
  if [ -z "${!var}" ]; then
    echo "The ${var#local_} has not been defined." >> "$LOG_FILE"
    exit 1
  fi
done
```

We'd use [Shell Parameter Expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html)

Reply:
Bash has arrays that could make this even simpler, or at least look cleaner.

Reply:
@nedvedba not quite sure what that implementation might look like?